pymc/distributions/transforms.py

```diff
@@ -82,7 +82,7 @@ def forward(self, value, *inputs):
         return y
 
     def log_jac_det(self, value, *inputs):
-        return at.sum(value[..., 1:], axis=-1)
+        return at.sum(value[..., 1:], axis=-1, keepdims=True)
```

Comment:
Setting `keepdims=True` will keep the last dimension while computing logprob for multivariate distributions. This behaviour can be a lot confusing to interpret when adding transforms for multivariate distributions.

Example:
```python
from pymc.distributions.logprob import joint_logpt

with pm.Model() as model:
    mv = pm.MvNormal(
        "mv",
        mu=[-3, -1, 1, 2],
        cov=np.eye(4),
        initval=np.random.randn(4),
    )

print(joint_logpt(mv, sum=False)[0].ndim)  # will output 0
```

After adding Ordered transform
```python
from pymc.distributions.logprob import joint_logpt

with pm.Model() as model:
    mv = pm.MvNormal(
        "mv",
        mu=[-3, -1, 1, 2],
        cov=np.eye(4),
        initval=np.random.randn(4),
        transform=pm.distributions.transforms.ordered,
    )

print(joint_logpt(mv, sum=False)[0].ndim)  # will output 1
```

I think there can be a simple check here: https://github.com/aesara-devs/aeppl/blob/65d25c39fa771809b4d992c9323f59e78f282cd5/aeppl/transforms.py#L372-L388

to squeeze out the last dimension in case when transforms are added for multivariate distributions.

cc @ricardoV94

Reply:
We might need to tell the transform what is the support dimensionality of the RV. Can be an internal variable like `n` in the transform that's used by the LKJCholeskyCov

Reply:
> We might need to tell the transform what is the support dimensionality of the RV.

If I get it right, we need to somehow pass `op.ndim_supp` to transforms to decide whether or not to keep last dimension. Seems like a simpler solution. 

Also, I think [SimplexTransform transform defined in AePPL](https://github.com/aesara-devs/aeppl/blob/main/aeppl/transforms.py#L289) needs to account for this change.

Reply:
We don't need to pass it to all transforms, we can instead initialize this transform with that information. Like we initialize Interval or LKJ transform with the specific information they need. This would follow a similar API and be less disruptive

In this case instead of having a single ordered already initialized, the user would initialize it (or us behind the scenes, like pm.interval does). It can be a kwarg that by default is 0 (for univariate RVs)

pseudo-code:
```python
class OrderedTransform(TransformRV):

  def __init__(self, ndim_supp):
    self.ndim_supp = ndim_supp

  def log_jac_det(self, value, *inputs):
    # use self.ndim_supp to return the right shape

def ordered(ndim_supp=0):
    return OrderedTransform(ndim_supp)
```

Reply:
Is there anything wrong with the Simplex? That seems to be vector specific transform, I don't think it would be used in other cases.

We can think of it as being a VectorSimplex (i.e., a Simplex for RVs with ndim_supp=1), which is what it is used for in PyMC. If we wanted to use it with other ndim_supp we could make it more flexible...

Reply:
Passing the required info to transforms during initialization indeed makes sense. @purna135 can you take up adding the proposed change in this PR itself?

> Is there anything wrong with the Simplex?

I noticed a test in `test_transforms.py` that applies Simplex transform to a batched univariate distribution. 

https://github.com/pymc-devs/pymc/blob/bae5087621323af3cb3cc95cf8d673da21ab4023/pymc/tests/test_transforms.py#L509-L526

This prompted me to make sure that even Simplex should not reduce the last dimension in case when applied to univariate distributions.

Reply:
Thank you, @Sayam753. I'll work on these suggested changes and let you know if I need any assistance.